src/scenes/UserGalleryPage/QRCode.tsx

```diff
@@ -0,0 +1,149 @@
+        dotsOptionsHelper: {
+          colorType: { single: true, gradient: false },
+          gradient: {
+            linear: true,
+            radial: false,
+            color1: '#6a1a4c',
+            color2: '#6a1a4c',
+            rotation: '0',
+          },
+        },
+        cornersSquareOptions: { type: '', color: '#000000' },
+        cornersSquareOptionsHelper: {
+          colorType: { single: true, gradient: false },
+          gradient: {
+            linear: true,
+            radial: false,
+            color1: '#000000',
+            color2: '#000000',
+            rotation: '0',
+          },
+        },
+        cornersDotOptions: { type: '', color: '#000000' },
+        cornersDotOptionsHelper: {
+          colorType: { single: true, gradient: false },
+          gradient: {
+            linear: true,
+            radial: false,
+            color1: '#000000',
+            color2: '#000000',
+            rotation: '0',
+          },
+        },
+        backgroundOptionsHelper: {
+          colorType: { single: true, gradient: false },
+          gradient: {
+            linear: true,
+            radial: false,
+            color1: '#ffffff',
+            color2: '#ffffff',
+            rotation: '0',
+          },
+        },
+      });
+
+      qrCode.append(ref.current);
+    }
+  }, [ref, username]);
+
+  const { showModal } = useModalActions();
+  const handleClick = useCallback(() => {
+    // Need a brief timeout so that the modal renders before rendering QR code. Otherwise the ref will not exist and renderQRCode cannot append the ref
+    setTimeout(() => {
+      renderQRCode();
+    }, 100);
```

Comment:
so we have a const for the modal transition time that's `ANIMATED_COMPONENT_TRANSITION_MS` which is 300ms. could we use that here? or would it be too slow?

Reply:
Technically I think the duration doesn't matter. (It also works with a `setTimeout` of 0) - the important part is that the modal begins to mount (no matter its animation status, it mounts instantly I think. So long as we wait a millisecond so that the modal is rendered before rendering the QR code, we're good